### PR TITLE
consul-template: 0.14.0 -> 0.18.0

### PR DESCRIPTION
**Pull request by pradeepchhetri**
###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by kamilchm — APPROVED** [review]

**Review by LnL7 — APPROVED** [review]

File: pkgs/tools/system/consul-template/default.nix [hunk 1]
```diff
@@ -11,6 +11,14 @@ buildGoPackage rec {
...
     owner = "hashicorp";
     repo = "consul-template";
-    sha256 = "15zsax44g3dwjmmm4fpb54mvsjvjf3b6g3ijskgipvhcy0d3j938";
+    sha256 = "1a1r7jwv0d4l8bcjal9chvr871hmw0ljbihgjqasp6gvjg0hfbx6";
+  };
+
+  meta = with stdenv.lib; {
+    homepage = https://github.com/hashicorp/consul-template/;
+    description = "Generic template rendering and notifications with Consul";
+    platforms = platforms.linux;
```
Comment (LnL7) [inline pkgs/tools/system/consul-template/default.nix]: Why is this linux only? It looks like it would work fine on other platforms to me.

**Comment by LnL7** [comment]
You should still define the platforms, use `platforms.unix` or `platforms.linux ++ platforms.darwin`

**Comment by pradeepchhetri** [comment]
@LnL7 Ohh ok. I made the change. If we don't define the platform, does it means `all platforms`

**Review by LnL7 — APPROVED** [review]

**Comment by LnL7** [comment]
No, it's undefined in that case. Nix won't fail when you try to build it, but hydra won't try to build it for example.

**Pull request merged**